test/e2e/replica_set_tls/replica_set_tls_test.go

```diff
@@ -0,0 +1,47 @@
+package replica_set_tls
+
+import (
+	"testing"
+
+	mdbv1 "github.com/mongodb/mongodb-kubernetes-operator/pkg/apis/mongodb/v1"
+	e2eutil "github.com/mongodb/mongodb-kubernetes-operator/test/e2e"
+	"github.com/mongodb/mongodb-kubernetes-operator/test/e2e/mongodbtests"
+	setup "github.com/mongodb/mongodb-kubernetes-operator/test/e2e/setup"
+	f "github.com/operator-framework/operator-sdk/pkg/test"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+func TestMain(m *testing.M) {
+	f.MainEntry(m)
+}
+
+func TestReplicaSetTLS(t *testing.T) {
+	ctx, shouldCleanup := setup.InitTest(t)
+	if shouldCleanup {
+		defer ctx.Cleanup()
+	}
+
+	mdb := e2eutil.NewTestMongoDB("mdb0")
+	mdb.Spec.Security.TLS = e2eutil.NewTestTLSConfig(false)
```

Comment:
I feel like we're going to need a better way of building MongoDB resources, perhaps a builder or even for now just a separate function to fully produce a TLS enabled MDB resource. I would also suggest maybe making the resource name a bit more meaningful, even just "mdb-tls" or "mdb-tls-upgrade"

Reply:
Have updated the tests to use the name "mdb-tls" instead. As the name is embedded in the certificate I've opted to use the same for both, otherwise we would need one certificate for each test.

We also agreed that a builder would be a nice addition. That work is tracked by a different ticket.